src/docs/arch-cache.dox

```diff
@@ -1,13 +1,110 @@
 /*! @arch_page arch-cache Cache
 
-Cache in WiredTiger is represented by the various shared data structures
-that make up in-memory Btrees and subordinate data structures.
+The WiredTiger cache is memory used to hold copies of recently accessed or modified data.
+WiredTiger reads btree pages into the cache on demand.  When the cache runs low on space, Eviction
```

Comment:
In the arch guide, we've been trying to use "Btree" instead of "btree".

Reply:
Done.